charts/nexus3/files/cleanup.groovy

```diff
@@ -2,19 +2,24 @@ import org.sonatype.nexus.cleanup.storage.CleanupPolicy;
 import org.sonatype.nexus.cleanup.storage.CleanupPolicyStorage;
 import groovy.json.JsonSlurper
 
+def cleanupPolicyStorage = container.lookup(CleanupPolicyStorage.class.getName())
+
 def params = new JsonSlurper().parseText(args)
-def policyStorage = container.lookup(CleanupPolicyStorage.class.getName());
 
-if (policyStorage.exists(params.name)) {
-  def cleanupPolicy = policyStorage.get(params.name);
-  cleanupPolicy.setNotes(params.notes);
-  cleanupPolicy.setFormat(params.format);
-  cleanupPolicy.setMode(params.mode);
-  cleanupPolicy.setCriteria(params.criteria);
-  policyStorage.update(cleanupPolicy);
+if (cleanupPolicyStorage.exists(params.name)) {
+  def existingPolicy = cleanupPolicyStorage.get(params.name);
+  existingPolicy.setNotes(params.notes);
+  existingPolicy.setFormat(params.format);
+  existingPolicy.setMode(params.mode);
+  existingPolicy.setCriteria(params.criteria);
+  cleanupPolicyStorage.update(existingPolicy);
 } else {
-  def policy = policyStorage.newCleanupPolicy()
-  policyStorage.add(policy);
+  def newPolicy = cleanupPolicyStorage.newCleanupPolicy()
```

Comment:
This is missing setting the name if the new cleanup policy

Reply:
This should now be fixed!